source/common/network/io_socket_handle_base_impl.cc

```diff
@@ -134,7 +134,10 @@ absl::optional<std::string> IoSocketHandleBaseImpl::interfaceName() {
 
   Api::InterfaceAddressVector interface_addresses{};
   const Api::SysCallIntResult rc = os_syscalls_singleton.getifaddrs(interface_addresses);
-  RELEASE_ASSERT(!rc.return_value_, fmt::format("getifaddrs error: {}", rc.errno_));
+  if (rc.return_value_ != 0) {
+    ENVOY_LOG(warn, "getifaddrs error: {}", rc.errno_);
```

Comment:
How often will this be logged if we're on a setup where getifaddrs is failing? If it's often, maybe use `ENVOY_LOG_PERIODIC`?

Reply:
Not entirely sure how often we expect this warning.  I didn't realize we had PERIODIC!  Given we aren't sure, maybe EVERY_POW_2 would be better?  I'll update to that.